SCons/Node/FSTests.py

```diff
@@ -1089,21 +1089,16 @@ def strip_slash(p, drive=drive):
             else:
                 dir_up_path = dir.up().get_internal_path()
 
-            assert dir.name == name, \
-                "dir.name %s != expected name %s" % \
-                (dir.name, name)
-            assert dir.get_internal_path() == path, \
-                "dir.path %s != expected path %s" % \
-                (dir.get_internal_path(), path)
-            assert str(dir) == path, \
-                "str(dir) %s != expected path %s" % \
-                (str(dir), path)
-            assert dir.get_abspath() == abspath, \
-                "dir.abspath %s != expected absolute path %s" % \
-                (dir.get_abspath(), abspath)
-            assert dir_up_path == up_path, \
-                "dir.up().path %s != expected parent path %s" % \
-                (dir_up_path, up_path)
+            with self.subTest():
```

Comment:
probably worth propagating the explanation strings in the asserts from the old code, it will make any failures easier to understand.